### ISSUE TITLE
[Bug] Wrong warnings about "Missing runtime dependencies"

### ISSUE THREAD

**Issue by tan-wei** [ScoopInstaller/Scoop#4796]
#### Current Behavior

I use scoop for some years and install many APPs. After the latest update (2022/03/10), it complains "Missing runtime dependencies" when run `scoop status`:
```
>_ scoop status
Scoop is up to date.
Missing runtime dependencies:
    'ack' requires 'perl'
    'elixir' requires 'erlang'
    'pandoc-crossref' requires 'pandoc'
    'z.lua' requires 'lua'
```
I can confirm that `perl`, `erlang`, `pandoc` and `lua` are all installed properly (and they are not updated this time):
```
>_ scoop list
Installed apps:

Name                              Version                                  Source
----                              -------                                  ------
0ad                               0.0.25b                                  games
1by1                              2.02                                     dodorz
7zip                              21.07                                    main
ag                                2.2.5                                    main
alacritty                         0.10.1                                   extras
alldup                            4.5.16                                   extras
ansicon                           1.89                                     main
appcompatibilityview              1.07                                     nirsoft-alternative
appcrashview                      1.35                                     nirsoft-alternative
appnetworkcounter                 1.47                                     nirsoft-alternative
appreadwritecounter               1.36                                     nirsoft-alternative
aria2                             1.36.0-1                                 main
ascii-sector                      0.7.2                                    games
atom                              1.60.0                                   extras
audacity                          3.1.3                                    extras
autoit                            3.3.16.0                                 main
autoit-script-editor              21.316.1639.0.0                          extras
base64                            0.94                                     dodorz
bind                              9.16.25                                  main
boostnote                         0.23.1                                   extras
boxes                             2.1.1                                    main
brave                             1.34.80-80                               extras
brotli                            1.0.9_4                                  main
cabal                             7.0.0                                    extras
cacert                            2022-02-01                               main
calibre-normal                    5.38.0                                   extras
captura                           8.0.0                                    extras
capture2text                      4.6.2                                    dodorz
chromedriver                      99.0.4844.51                             main
clash-for-windows                 0.19.11                                  dorado
clementine                        1.3.1                                    extras
CLion                             2021.3.3-213.6777.58                     jetbrains
cloc                              1.92                                     main
clojure                           1.10.3.1087                              scoop-clojure
clumsy                            0.2                                      extras
cpdf                              v2.5.1                                   rasa
cports                            2.65                                     nirsoft-alternative
cppcheck                          2.7                                      main
croc                              9.5.2                                    main
cscope                            15.8a                                    main
csvfileview                       2.55                                     nirsoft-alternative
csview                            1.0.1                                    main
cuetools                          2.2.1                                    extras
curl                              7.82.0                                   main
czkawka                           4.0.0                                    main
dbeaver                           22.0.0                                   extras
delta                             0.12.1                                   main
dependencies                      1.11.1                                   extras
depends                           2.2                                      extras
diagram-designer                  1.29.5                                   extras
diff-pdf                          0.5                                      extras
diffpdf                           2.1.3                                    extras
dirhash                           1.23.0                                   main
discord                           0.0.311                                  extras
djvulibre                         3.5.28-4.12                              main
dngrep                            3.0.29.0                                 extras
dnsbench                          1.3.6688.0                               rasa
dnsjumper                         2.2                                      extras
dnsquerysniffer                   1.85                                     nirsoft-alternative
dog                               0.1.0                                    main
dotnet-sdk                        6.0.201                                  main
DoubleCommander                   1.0.4                                    dodorz
draw.io                           16.5.1                                   extras
edgedriver                        100.0.1188.0                             main
electerm                          1.18.5                                   dorado
electrum                          4.1.5                                    extras
element                           1.10.6                                   extras
emacs                             27.2                                     extras
espanso                           0.7.3                                    main
etcd                              3.5.2                                    main
falcon-sql-client                 4.1.0                                    extras
fastcopy                          4.1.1                                    extras
feishu                            5.6.9                                    dorado
fiddler                           5.0.20211.51073                          extras
filezilla                         3.58.0                                   extras
flameshot                         11.0.0                                   extras
fork                              1.70.0                                   extras
fscapture                         9.7                                      extras
fselect                           0.8.0                                    dodorz
FullEventLogView                  1.67                                     nirsoft
fvim                              0.3.489                                  extras
fzf                               0.29.0                                   main
geany                             1.38                                     extras
geckodriver                       0.30.0                                   main
geth                              1.10.16                                  dorado
gh                                2.5.2                                    main
git-annex                         10.20220128                              main
git-bug                           0.7.2                                    main
gitea                             1.16.3                                   main
gitify                            4.3.1                                    extras
gitignore                         0.2018.07.25                             main
glitter                           1.5.16                                   main
glow                              1.4.1                                    main
goland                            2021.3.3-213.6777.51                     jetbrains
goneovim                          0.5.1                                    extras
gource                            0.51                                     main
grpcurl                           1.8.6                                    main
gsudo                             1.1.0                                    main
hashcat                           6.2.5                                    main
Helix                             0.6.0                                    dodorz
hexchat                           2.16.1                                   extras
hexed                             1.0.32                                   main
hexyl                             0.9.0                                    main
highlight                         4.1                                      main
hjson                             3.0.0                                    main
httpstat                          1.0.0                                    main
hyper                             3.2.0                                    extras
ImageFinder                       1.003                                    dodorz
imdone                            1.27.1                                   extras
imhex                             1.16.2                                   extras
IntelliJ-IDEA-Ultimate            2021.3.2-213.6777.52                     jetbrains
iperf3                            3.1.3                                    main
jellyfin                          10.7.7                                   extras
joplin                            2.7.13                                   extras
jq                                1.6                                      main
jx                                3.2.263                                  main
kate                              21.12.3-1590                             versions
keypirinha                        2.26                                     extras
klogg                             20.12.0.813                              extras
LANDrop                           0.4.0                                    dodorz
lf                                r26                                      main
libreoffice                       7.2.5                                    extras
licecap                           1.28                                     extras
lighttpd                          1.4.49-1                                 extras
lite                              1.11                                     extras
lite-xl                           2.0.5                                    extras
livetcpudpwatch                   1.41                                     nirsoft-alternative
logseq                            0.6.2                                    C:\Users\Winterreise\scoop\buckets\extras...
logstalgia                        1.0.9                                    extras
logstash                          6.8.23                                   extras
love                              11.4                                     extras
magicavoxel                       0.99.7.0                                 extras
mancy                             3.2.0                                    extras
markdown-monster                  1.28.5                                   extras
MarkDownEditor                    0.4.0                                    dodorz
marktext                          0.17.1                                   extras
meld                              3.20.4                                   extras
memreduct                         3.4                                      extras
micro                             2.0.10                                   main
mist                              0.11.1                                   extras
motrix                            1.6.11                                   extras
mp3tag                            3.12                                     extras
mpv                               0.34.0                                   extras
msbuild-structured-log-viewer     2.1.627                                  extras
multrin                           1.3.0                                    extras
musicbrainz-picard                2.7.3                                    dorado
neovim-nightly                    0.7.0-dev-1217-g205b3765f                versions
netcat                            1.12                                     main
nethack                           3.6.6                                    games
networklatencyview                1.66                                     nirsoft-alternative
networkopenedfiles                1.50                                     nirsoft-alternative
networktrafficview                2.41                                     nirsoft-alternative
nomacs                            3.16.1709                                extras
notion                            2.0.23                                   dorado
nu                                0.44.0                                   main
obsidian                          0.13.31                                  C:\Users\Winterreise\scoop\buckets\extras...
oh-my-posh                        7.35.0                                   main
okular                            21.12.3-946                              versions
oni                               0.3.9                                    extras
open-log-viewer                   1.5.1                                    extras
paket                             7.0.2                                    main
PandaOCR                          2.71                                     dodorz
pcregrep                          10.20                                    main
pdf-xchange-editor                9.2.359.0                                extras
pdfcpu                            0.3.13                                   main
pe-bear                           0.5.5.3                                  extras
peco                              0.5.10                                   main
phantomjs                         2.1.1                                    main
picpick                           5.2.1                                    dodorz
poppler                           22.01.0-0                                main
postman                           9.14.0                                   extras
potplayer                         220302                                   extras
powertoys-np                      0.56.2                                   nonportable
processtcpsummary                 1.16                                     nirsoft-alternative
putty                             0.76                                     extras
PyCharm-Professional              2021.3.2-213.6777.50                     jetbrains
q-dir                             10.63                                    extras
qpdf                              10.6.2                                   main
qtscrcpy                          1.8.1                                    dorado
RavenReader                       1.0.71                                   dodorz
redis                             5.0.14.1                                 main
rehex                             0.4.1                                    extras
restic                            0.12.1                                   main
rga                               0.9.6                                    main
ripgrep                           13.0.0                                   main
Rubymine                          2021.3.2-213.6777.43                     jetbrains
rustup                            1.24.3                                   main
scc                               3.0.0                                    main
SciTE                             5.2.1                                    dodorz
screentogif                       2.36                                     extras
sdl2                              2.0.20                                   extras
shadowsocks                       4.4.1.0                                  extras
shadowsocksr-csharp               4.9.2                                    extras
shellexview                       2.01                                     nirsoft
shexview                          2.01                                     nirsoft-alternative
sigil                             1.9.1                                    extras
simplenote                        2.21.0                                   extras
siyuan                            1.8.9                                    dorado
smartsniff                        2.29                                     nirsoft
smsniff                           2.29                                     nirsoft-alternative
snipaste                          1.16.2                                   extras
so                                0.4.6                                    extras
solidity                          0.8.12                                   dorado
sourcetrail                       2021.4.19                                extras
sourcetree                        3.4.8                                    extras
speedtest-cli                     1.1.1                                    dorado
sqlitebrowser                     3.12.2                                   extras
SQLiteExpert                      5.4.9                                    dodorz
sqlitestudio                      3.3.3                                    extras
starship                          1.4.0                                    main
sudo                              0.2020.01.26                             main
sumatrapdf                        3.3.3                                    dodorz
syncthing                         1.19.1                                   main
tabby                             1.0.173                                  dodorz
TablacusExplorer                  22.3.7                                   dodorz
tageditor                         3.6.1                                    extras
taskschedulerview                 1.70                                     nirsoft
tcping                            0.39                                     dorado
tcplogview                        1.35                                     nirsoft-alternative
telnet                            msys-inetutils-1.7-1                     main
tesseract                         5.0.1.20220118                           main
tesseract-languages               4.1.0                                    main
texlab                            3.3.2                                    main
texmacs                           2.1.1                                    extras
texmaker                          5.1.2                                    extras
texstudio                         4.2.2                                    extras
text-grab                         2.5.1                                    extras
texteditorpro                     20.1.0                                   extras
thunderbird                       91.7.0                                   extras
time                              0.2018.07.25                             main
time-it                           1.2                                      rasa
tldr                              0.6.1                                    main
todolist                          8.0.24                                   extras
trafficmonitor                    1.83                                     dorado
typora                            1.1.5                                    extras
ugrep                             3.7.4                                    main
universal-ctags                   2020-10-11                               extras
usql                              0.9.5                                    main
uutils-coreutils                  0.0.12                                   main
vagrant                           2.2.19                                   main
ventoy                            1.0.70                                   extras
vim-tux                           8.2.4525                                 C:\Users\Winterreise\scoop\buckets\versio...
vimtutor                          0.2018.07.25                             main
virt-viewer                       11.0-1.0                                 extras
virtualbox-with-extension-pack-np 6.1.32                                   nonportable
vlc                               3.0.16                                   extras
vncviewer                         6.22.207                                 extras
vnote                             3.12.888                                 extras
watchexec                         1.18.3                                   main
WeChat                            3.5.0                                    dodorz
wesnoth                           1.16.2                                   games
wezterm                           20220101-133340-7edc5b5a                 extras
wget                              1.21.2                                   main
win11dragdroptaskbarfix           2.3.0.0                                  extras
win32-disk-imager                 1.0.0                                    extras
wkhtmltopdf                       0.12.6-1                                 main
wxhexeditor                       0.24-Beta                                dodorz
xsv                               0.13.0                                   main
yarn                              1.22.17                                  main
Yosoro                            1.1.7                                    dodorz
you-get                           0.4.1555                                 dodorz
youtube-dl                        2021.12.17                               main
youtube-dl-gui                    1.8.3                                    extras
youtube-dlc                       2020.11.11-3                             versions
yq                                4.21.1                                   main
yt-dlp                            2022.03.08.1                             main
zettlr                            2.2.4                                    extras
zulip                             5.8.1                                    extras
3270-NF                           2.1.0                                    nerd-fonts
3270-NF-Mono                      2.1.0                                    nerd-fonts
7zip                              21.07                                    main
ack                               3.5.0                                    main
Agave-NF                          2.1.0                                    nerd-fonts
Agave-NF-Mono                     2.1.0                                    nerd-fonts
AnonymousPro-NF                   2.1.0                                    nerd-fonts
AnonymousPro-NF-Mono              2.1.0                                    nerd-fonts
Anuphan                           3.002                                    nerd-fonts
Arimo-NF                          2.1.0                                    nerd-fonts
Arimo-NF-Mono                     2.1.0                                    nerd-fonts
BigBlueTerminal-NF                2.1.0                                    nerd-fonts
BigBlueTerminal-NF-Mono           2.1.0                                    nerd-fonts
BitstreamVeraSansMono-NF          2.1.0                                    nerd-fonts
BitstreamVeraSansMono-NF-Mono     2.1.0                                    nerd-fonts
Cascadia-Code                     2111.01                                  nerd-fonts
ccache                            4.6                                      main
ChezScheme                        9.5.6                                    dodorz
chromium                          99.0.4844.51-r961656                     extras
clink                             1.3.9                                    main
cmake                             3.22.3                                   main
cmder                             1.3.19                                   main
cormanlisp                        3.1.2b                                   main
dark                              3.11.2                                   main
DejaVuSansMono-NF                 2.1.0                                    nerd-fonts
DejaVuSansMono-NF-Mono            2.1.0                                    nerd-fonts
Delugia-Nerd-Font                 2111.01                                  nerd-fonts
Delugia-Nerd-Font-Book            2111.01                                  nerd-fonts
Delugia-Nerd-Font-Complete        2111.01                                  nerd-fonts
deno                              1.19.2                                   main
dmd                               2.098.1                                  main
doxygen                           1.9.2                                    main
dvc                               2.9.5                                    main
elixir                            1.13.3                                   main
erlang                            24.2.2                                   main
esbuild                           0.14.25                                  dorado
FantasqueSansMono-NF              2.1.0                                    nerd-fonts
FantasqueSansMono-NF-Mono         2.1.0                                    nerd-fonts
ffmpeg                            5.0                                      main
FiraCode                          6.2                                      nerd-fonts
FiraMono-NF                       2.1.0                                    nerd-fonts
FiraMono-NF-Mono                  2.1.0                                    nerd-fonts
Font-Awesome                      6.0.0                                    nerd-fonts
ghostscript                       9.55.0                                   main
git                               2.35.1.windows.2                         main
git-filter-repo                   2.34.0                                   main
gitui                             0.20.1                                   main
go                                1.17.8                                   main
Go-Mono-NF                        2.1.0                                    nerd-fonts
Go-Mono-NF-Mono                   2.1.0                                    nerd-fonts
graphviz                          2.49.3                                   main
gsudo                             1.1.0                                    main
Hack-NF                           2.1.0                                    nerd-fonts
Hasklig                           1.2                                      nerd-fonts
Hasklig-NF                        2.1.0                                    nerd-fonts
Hermit-NF                         2.1.0                                    nerd-fonts
Hermit-NF-Mono                    2.1.0                                    nerd-fonts
IBMPlexMono-NF                    2.1.0                                    nerd-fonts
IBMPlexMono-NF-Mono               2.1.0                                    nerd-fonts
icaros-np                         3.2.1                                    nonportable
imagemagick                       7.1.0-27                                 main
inconsolata-NF                    2.1.0                                    nerd-fonts
inconsolataGo-NF                  2.1.0                                    nerd-fonts
inconsolataLGC-NF                 2.1.0                                    nerd-fonts
innounp                           0.50                                     main
Iosevka-NF                        2.1.0                                    nerd-fonts
janet                             1.20.0                                   main
JetBrains-Mono                    2.242                                    nerd-fonts
julia                             1.7.2                                    main
lazygit                           0.33                                     extras
lessmsi                           1.10.0                                   main
llvm                              13.0.1                                   main
lua                               5.4.3                                    main
Meslo-NF                          2.1.0                                    nerd-fonts
Meslo-NF-Mono                     2.1.0                                    nerd-fonts
mono                              6.12.0.107                               main
Monofur-NF                        2.1.0                                    nerd-fonts
Monofur-NF-Mono                   2.1.0                                    nerd-fonts
Monoid-NF                         2.1.0                                    nerd-fonts
Monoid-NF-Mono                    2.1.0                                    nerd-fonts
msmpi                             10.1.1                                   main
msys2                             2022-01-28                               main
nb                                6.9.5                                    dodorz
ninja                             1.10.2                                   main
Noto-Sans-Font                    2017.10.25                               dodorz
nsis                              3.08                                     extras
nvm                               1.1.9                                    main
Open-Sans                         nightly-20211202                         nerd-fonts
openjdk11                         11.0.2-9                                 java
openssh                           8.6.0.0p1                                main
openssl                           3.0.1                                    main
operator-mono-fonts               1370287d8e31ef55cd94c157fa9dbed189f28bae dodorz
pandoc                            2.17.1.1                                 main
pandoc-crossref                   0.3.12.2                                 main
perl                              5.32.1.1                                 main
ProFont-NF                        2.1.0                                    nerd-fonts
ProFont-NF-Mono                   2.1.0                                    nerd-fonts
protobuf                          3.19.4                                   extras
py                                1.0.1.6                                  dodorz
python                            3.10.2                                   main
racket                            8.4                                      main
RobotoMono-NF                     2.1.0                                    nerd-fonts
RobotoMono-NF-Mono                2.1.0                                    nerd-fonts
ruby                              3.1.1-1                                  main
rust-analyzer                     2022-02-28                               main
SarasaGothic                      0.36.0                                   nerd-fonts
SarasaGothic-SC                   0.36.0                                   nerd-fonts
sbcl                              2.2.2                                    main
sbt                               1.6.2                                    main
scala                             3.1.1                                    main
sccache                           0.2.14                                   main
Setofont                          6.20                                     nerd-fonts
ShareTechMono-NF                  2.1.0                                    nerd-fonts
ShareTechMono-NF-Mono             2.1.0                                    nerd-fonts
Source-Han-Serif-SC               2.001                                    nerd-fonts
SourceCodePro-NF                  2.1.0                                    nerd-fonts
SourceCodePro-NF-Mono             2.1.0                                    nerd-fonts
sqlite                            3.38.0                                   main
stack                             2.7.3                                    main
sudo                              0.2020.01.26                             main
swig                              4.0.2                                    main
tcc                               0.9.27                                   main
Victor-Mono                       1.5.2                                    nerd-fonts
VictorMono-NF                     2.1.0                                    nerd-fonts
VictorMono-NF-Mono                2.1.0                                    nerd-fonts
vulkan                            1.3.204.1                                main
winflexbison                      2.5.25                                   main
winpty                            0.4.3                                    extras
x64dbg                            2022-03-02_15-55                         extras
z.lua                             1.8.14                                   main
zig                               0.9.1                                    main
```

#### Expected Behavior
Does not complain "Missing runtime dependencies" because they are all installed.

### System details

**Windows version:** 10

**OS architecture:** 64bit

**PowerShell version:** 5.1.19041.1320

**Comment by tan-wei** [comment]
It seems the dependencies scoop complains are all installed in global (with `-g`).

**Comment by tan-wei** [comment]
Waiting for the commit is merged into `main`.

**Comment by rashil2000** [comment]
Tracked in https://github.com/ScoopInstaller/Scoop/issues/4819

**Comment by wisdomtooth** [comment]
> It seems the dependencies scoop complains are all installed in global (with `-g`).

I can confirm that. These are the warnings I was getting:

```
Missing runtime dependencies:
    'scribus-dev' requires 'ghostscript'
    'superputty' requires 'putty'
    'Wget' requires 'cacert'
    'winsshterm' requires 'putty'
```

Installing these dependencies non-globally eliminated the warnings (though I generally prefer installing everything global, as I'm the only one using this machine, and want to see all the apps I install regardless of which profile I'm in--if for whatever reason I need to log into another one).

**Comment by kamenminkov** [comment]
Similar story here. I've got `everything` and `everything-cli` both installed globally:

```
> scoop list | grep everything
everything                      1.4.1.1015       extras     2022-05-02 11:05:49 Global install
everything-cli                  1.1.0.23         main       2022-05-02 15:05:04 Global install
```
... and scoop complains as follows:

```
Missing runtime dependencies:
    'everything-cli' requires 'extras/everything'
```

If I install those locally, they work fine from Scoop's point of view (i.e. no such warning), but cause problems by not always wanting to start with Windows or running as admin (hence why I installed them globally).
